### ISSUE TITLE
What is the best way to max pool top k elements instead of 1?

### ISSUE THREAD

**Issue by zhang8473** [tensorflow/tensorflow#1670]
I would like to know if there is a convenient way to pool top k elements instead of just one.

The tf.nn.top_k seems not for this kind of job, because for each window, I want the top k elements.

**Comment by keveman** [comment]
TensorFlow doesn't have this operation currently, and doing it with existing ops is probably going to be clunky. Do you mind describing the operation slightly more formally, so that it can act as a specification for someone who can pick it up to implement it? Also, please post it to StackOverflow, in case someone in the larger community has an idea to implement it using existing TensorFlow operations.

**Comment by zhang8473** [comment]
Now the max pooling function in tensorflow is
<pre>
tf.nn.max_pool(value, ksize, strides, padding, name=None)
Returns:
A Tensor with type tf.float32. The max pooled output tensor.
</pre>

I would like to have an extend version of max_pool, like
<pre>
tf.nn.top_k_pool(value, ksize, strides, padding, k=1, name=None)

Performs the top k pooling on the input.

Args:

value: A 4-D Tensor with shape [batch, height, width, channels] and type tf.float32.
ksize: A list of ints that has length >= 4. The size of the window for each dimension of the input tensor.
strides: A list of ints that has length >= 4. The stride of the sliding window for each dimension of the input tensor.
padding: A string, either 'VALID' or 'SAME'. The padding algorithm.
k: 0-D int32 Tensor. Number of top elements to look in each pool.
name: Optional name for the operation.
Returns:

A Tensor with type tf.float32. The max pooled output tensor. There will be an additional dimension saving the top k values.
</pre>

I know that I can expend the tensorflow operation following https://www.tensorflow.org/versions/r0.7/how_tos/adding_an_op/index.html

I would like to know if there is an easier way to achieve that.

**Comment by ljm52519** [comment]
Have you solved this issue? I want to max pool top k elements too,but tf.nn.top_k only search on the last dimension.
